includes/tracks/class-wc-site-tracking.php

```diff
@@ -100,11 +128,12 @@ public static function add_empty_tracking_function() {
 	 */
 	public static function init() {
 
-		if ( ! self::is_tracking_enabled() ) {
-
-			// Define window.wcTracks.recordEvent in case there is an attempt to use it when tracking is turned off.
-			add_filter( 'admin_footer', array( __CLASS__, 'add_empty_tracking_function' ), 24 );
+		// Define window.wcTracks.recordEvent in case it is enabled client-side.
+		self::register_scripts();
+		add_filter( 'admin_footer', array( __CLASS__, 'add_tracking_function' ), 24 );
 
+		if ( ! self::is_tracking_enabled() ) {
+			add_filter( 'admin_footer', array( __CLASS__, 'add_enable_tracking_function' ), 24 );
 			return;
```

Comment:
[not a blocker]. just curious, since we return here if tracking is not enabled, do you know if files that are not loaded will cause us to miss on events if tracking is enabled on the fly? as far as I can see, all these files track events on different screens so its likely that we would have refreshed by then.

Reply:
Good question.  All these pages will invoke a refresh prior to viewing so they should be covered.

The only potential exception would be if the embedded wc-admin notifications bar, which runs on some of those pages, would contain a note or some action to opt-in to tracking, we could miss events.

However, that functionality doesn't currently exist.  We'll work on a fallback for edge cases like this in wc-admin in the event that we ever add in something like this and future-proof these pages a bit.